expression/builtin_other.go

```diff
@@ -615,13 +615,30 @@ func (b *builtinValuesStringSig) evalString(_ chunk.Row) (string, bool, error) {
 	if row.IsEmpty() {
 		return "", true, errors.New("Session current insert values is nil")
 	}
-	if b.offset < row.Len() {
-		if row.IsNull(b.offset) {
-			return "", true, nil
+	if b.offset >= row.Len() {
+		return "", true, errors.Errorf("Session current insert values len %d and column's offset %v don't match", row.Len(), b.offset)
+	}
+
+	if row.IsNull(b.offset) {
+		return "", true, nil
+	}
+
+	// Specially handle the ENUM/SET/BIT input value.
+	if retType := b.getRetTp(); retType.Hybrid() {
+		val := row.GetDatum(b.offset, retType)
+		res, err := val.ToString()
+		if err != nil {
+			return "", true, err
 		}
-		return row.GetString(b.offset), false, nil
+
+		resLen := len([]rune(res))
+		if b.ctx.GetSessionVars().StmtCtx.PadCharToFullLength && retType.Tp == mysql.TypeString && resLen < retType.Flen {
```

Comment:
Since the `retType` is hybird, will it be `TypeString`?

Reply:
no. updated.